### PR TITLE
Added tower of hanoi problem in C++

### PR DESCRIPTION
**Pull request by SAY-droid427**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Tower of Hanoi in C++

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

https://github.com/TesseractCoding/NeoAlgo/issues/2664

### CODE REVIEW

File: C-Plus-Plus/cp/tower_of_hanoi.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/* Tower Of Hanoi
+Tower of Hanoi is a mathematical puzzle where we have three rods and n disks. The objective of the puzzle is to move the entire stack to another rod using the below rules:
+*Only one disk can be moved at a time.
+*Each move consists of taking the upper disk from one of the stacks and placing it on top of another stack i.e. a disk can only be moved if it is the uppermost disk on a stack.
+*No disk must be placed on top of a smaller disk.
+*/
+#include <bits/stdc++.h>
+using namespace std;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/tower_of_hanoi.cpp:8]: Add a new line after this.

File: C-Plus-Plus/cp/tower_of_hanoi.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* Tower Of Hanoi
+Tower of Hanoi is a mathematical puzzle where we have three rods and n disks. The objective of the puzzle is to move the entire stack to another rod using the below rules:
+*Only one disk can be moved at a time.
+*Each move consists of taking the upper disk from one of the stacks and placing it on top of another stack i.e. a disk can only be moved if it is the uppermost disk on a stack.
+*No disk must be placed on top of a smaller disk.
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/tower_of_hanoi.cpp:6]: Break into subpoints and make it concise.

File: C-Plus-Plus/cp/tower_of_hanoi.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+{
+    if (n == 1)
+    {
+        cout << "Move disk 1 from rod " << from_rod << " to rod " << to_rod << endl;
+        return;
+    }
+    towerOfHanoi(n - 1, from_rod, intermediate_rod, to_rod);
+    cout << "Move disk " << n << " from rod " << from_rod << " to rod " << to_rod << endl;
+    towerOfHanoi(n - 1, intermediate_rod, to_rod, from_rod);
+}
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/tower_of_hanoi.cpp:19]: Add a new line after this.

File: C-Plus-Plus/cp/tower_of_hanoi.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+int main()
+{
+    int n;
+    cin >> n;
+    towerOfHanoi(n, 'A', 'C', 'B');
+}
+/*
+Time Complexity: O(n)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/tower_of_hanoi.cpp:27]: Add Time and Space Complexity at the end of the Code   
Also Time complexity of Tower of Hanoi problem using recursion is O( 2^n - 1)  
How it is O(n)?

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/tower_of_hanoi.cpp (hunk 4 above)
Comment (SAY-droid427) [reply C-Plus-Plus/cp/tower_of_hanoi.cpp:27]: @ankitaggarwal23 Sorry for not being able to commit the changes earlier. I have created a new pull request https://github.com/TesseractCoding/NeoAlgo/pull/3980/

**Comment by ankitaggarwal23** [comment]
@SAY-droid427 Problem already exists in the Mathematics section #3116

**Pull request closed without merging**